### PR TITLE
implement Array.Extra.memeber function

### PR DESCRIPTION
**Pull request by turboMaCk**
Also adds benchmarks for candidate implementations:

* converting to List seems to be fastest for long Arrays
* folding seems to be fastest in most common case
* manual recursion with bailout is slower then rest

This is benchmark result from one of my runs in FireFox:

image

It almost seems we could just convert to List to do this but it feels wrong.

### CODE REVIEW

**Comment by lue-bird** [comment]
Would be nice to add a fuzz-test like
```elm
memberTest : Test
memberTest =
    Test.fuzz
        (Fuzz.constant
            (\before after ->
                { before = before, after = after }
            )
            |> Fuzz.andMap (Fuzz.array Fuzz.int)
            |> Fuzz.andMap (Fuzz.array Fuzz.int)
        )
        "member"
        (\{ before, after } ->
            Array.append
                (before |> Array.push 123456)
                after
                |> Array.Extra.member 123456
                |> Expect.equal True
        )
```

**Comment by turboMaCk** [comment]
Agree! In meantime are you able to approve workflow @lue-bird ?

**Comment by lue-bird** [comment]
Sorry, I'm [still waiting to become accepted as maintainer](https://github.com/elm-community/Manifesto/pull/102)

**Comment by turboMaCk** [comment]
Fuzz test added. I also took a liberty and [updated .gitignore](https://github.com/elm-community/array-extra/pull/32/commits/7927c090fd28f8e7f595e222da10e710238d4ec6) because of my OCD side which can't stand seeing generated things in diff.

File: .gitignore
Comment (turboMaCk) [inline .gitignore:4]: I hit this one with benchmarks but I think it's good to have this global rule.

File: benchmarks/src/Candidates.elm
Comment (turboMaCk) [inline benchmarks/src/Candidates.elm:238]: jsut realized this could be probably "simplified".

```suggestion
            i == item || arrayMemberRecHelp (n + 1) item array
               
```

Comment (turboMaCk) [reply benchmarks/src/Candidates.elm:238]: What I'm not sure though is if this could not break tail call optimizer.

Comment (lue-bird) [reply benchmarks/src/Candidates.elm:238]: [I think it does break TCO](https://jfmengels.net/tail-call-optimization/#calls-appearing-in--or--conditions)

**Comment by jonboiser** [comment]
@turboMaCk @lue-bird I'm still following up with the elm-community admins to get lue-bird admin permissions to this repo. As far as this PR goes, though, I'm happy to merge as soon as lue-bird gives an approving review 👍

File: benchmarks/src/Benchmarks.elm
Comment (jonboiser) [inline benchmarks/src/Benchmarks.elm:120]:
```suggestion
            , ( "with List.member", listMember )
```

File: tests/Tests.elm
Comment (jonboiser) [inline tests/Tests.elm:447]:
```suggestion
                "doesn't exist"
```

**Review by lue-bird — APPROVED** [review]
A last consideration:

I remember not implementing
```elm
member needle = any ((==) needle)
```
in favor of the `any` version to make a simpler API.

Adding `member` might be good for those searching for that exact operation while allowing the library to choose the more performant version (operator currying vs lambda).

Let's maybe benchmark these  `member` implementations using `any`, too, since I think the direct fold version will only be marginally faster.

**Comment by jonboiser** [comment]
@lue-bird I sent an invite to be a maintainer on the repo. Feel free to merge this when you feel it's ready.

I made a wiki page on how to make a release [here](https://github.com/elm-community/array-extra/wiki#how-to-make-a-release), if you're not familiar with the process.

**Pull request merged**
